### PR TITLE
prune prefix or

### PR DESCRIPTION
**Pull request by dentiny**
```
memory D COPY (
             SELECT range::INTEGER AS x, 'p' || lpad(range::VARCHAR, 6, '0') AS s
             FROM range(8192)
         ) TO '/tmp/expr_prune_prefix_or.parquet' (FORMAT PARQUET, ROW_GROUP_SIZE 2048);
memory D EXPLAIN ANALYZE SELECT count(*), sum(x)
         FROM '{TEST_DIR}/expr_prune_prefix_or.parquet'
         WHERE s LIKE 'p0000%';
IO Error:
No files found that match the pattern "{TEST_DIR}/expr_prune_prefix_or.parquet"
memory D EXPLAIN ANALYZE SELECT count(*), sum(x)
         FROM '/tmp/expr_prune_prefix_or.parquet'
         WHERE s LIKE 'p0000%';
╭─ Summary ───────────╮
│ Total Time: 0.0033s │
│ Data Read:  33.9 kB │
╰─────────────────────╯
╭─ Ungrouped Aggregate ─────────────╮
│ Aggregates: count_star(), sum(#0) │
│ 1 row                         0µs │
╰─────────────────┒┎────────────────╯
╭─ Projection ────┚┖────────────────╮
│ Projections: x                    │
│ 100 rows                      0µs │
╰─────────────────┒┎────────────────╯
╭─ Table Scan ────┚┖────────────────╮
│ Function: PARQUET_SCAN            │
│ Projections: x                    │
│ Filters:                          │
│ (s >= 'p0000') AND (s < 'p0001')  │
│ Total Files Read: 1               │
│ Filename(s):                      │
│ /tmp/expr_prune_prefix_or.parquet │
│ Row Groups Scanned: 1 / 4         │
│ 100 rows                      0µs │
╰───────────────────────────────────╯

type .web to open the query profile in a browser
memory D EXPLAIN ANALYZE SELECT count(*), sum(x)
                  FROM '/tmp/expr_prune_prefix_or.parquet'
                  WHERE s LIKE 'p0050%';
╭─ Summary ───────────╮
│ Total Time: 0.0009s │
│ Data Read:  34.0 kB │
╰─────────────────────╯
╭─ Ungrouped Aggregate ─────────────╮
│ Aggregates: count_star(), sum(#0) │
│ 1 row                         0µs │
╰─────────────────┒┎────────────────╯
╭─ Projection ────┚┖────────────────╮
│ Projections: x                    │
│ 100 rows                      0µs │
╰─────────────────┒┎────────────────╯
╭─ Table Scan ────┚┖────────────────╮
│ Function: PARQUET_SCAN            │
│ Projections: x                    │
│ Filters:                          │
│ (s >= 'p0050') AND (s < 'p0051')  │
│ Total Files Read: 1               │
│ Filename(s):                      │
│ /tmp/expr_prune_prefix_or.parquet │
│ Row Groups Scanned: 1 / 4         │
│ 100 rows                      0µs │
╰───────────────────────────────────╯

type .web to open the query profile in a browser
memory D EXPLAIN ANALYZE SELECT count(*), sum(x)
                  FROM '/tmp/expr_prune_prefix_or.parquet'
                  WHERE s LIKE 'p0000%' OR s LIKE 'p0050%';
╭─ Summary ───────────╮
│ Total Time: 0.0024s │
│ Data Read:  86.6 kB │
╰─────────────────────╯
╭─ Ungrouped Aggregate ────────────────────╮
│ Aggregates: count_star(), sum(#0)        │
│ 1 row                                0µs │
╰─────────────────────┒┎───────────────────╯
╭─ Projection ────────┚┖───────────────────╮
│ Projections: x                           │
│ 200 rows                             0µs │
╰─────────────────────┒┎───────────────────╯
╭─ Table Scan ────────┚┖───────────────────╮
│ Function: PARQUET_SCAN                   │
│ Projections: x                           │
│ Filters:                                 │
│ prefix(s, 'p0000') OR prefix(s, 'p0050') │
│ Total Files Read: 1                      │
│ Filename(s):                             │
│ /tmp/expr_prune_prefix_or.parquet        │
│ Row Groups Scanned: 4 / 4                │
│ 200 rows                             0µs │
╰──────────────────────────────────────────╯
```

---

> [!NOTE]
> **Medium Risk**
> Changes query optimization and filter semantics at the statistics layer; incorrect pruning would drop valid rows, though behavior is guarded by conservative `NO_PRUNING_POSSIBLE` fallbacks and extensive tests.
> 
> **Overview**
> Adds **statistics-based row-group pruning** for `prefix(column, constant)` so Parquet and other scans can skip groups that cannot match, including when several prefix branches are combined with **OR** (e.g. `LIKE 'p0000%' OR LIKE 'p0050%'`).
> 
> The `prefix` scalar function now registers a **filter-prune callback** that uses string column stats (nullability, max length, min/max) and a byte-wise **successor** of the constant prefix (`FindNextPrefix`) to prove a group is always false—or, for an empty prefix with no nulls, always true. Non-constant second arguments, null prefixes, and missing stats leave pruning disabled. **`StringStats::CompareStringStats`** is exposed publicly so prefix pruning shares the same truncated/exact min-max comparison as zonemap checks.
> 
> New Parquet expression-filter tests cover OR pruning, corner cases (empty/NULL/long prefixes, UTF-8 byte boundaries), and verify row-group scan counts via `EXPLAIN ANALYZE`.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 83cc9645e6a5956800b9231bb9ac8c13ff92479a. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: src/function/scalar/string/prefix.cpp [hunk 1]
```diff
@@ -59,13 +65,74 @@ struct PrefixOperator {
...
+	auto &children = input.function.GetChildren();
+	if (children.size() != 2 || children[1]->GetExpressionType() != ExpressionType::VALUE_CONSTANT) {
+		return FilterPropagateResult::NO_PRUNING_POSSIBLE;
+	}
+
+	auto column_stats = input.ChildStats(0);
+	if (!column_stats || column_stats->GetStatsType() != StatisticsType::STRING_STATS) {
+		return FilterPropagateResult::NO_PRUNING_POSSIBLE;
+	}
+	if (!column_stats->CanHaveNoNull()) {
```
Comment (dentiny) [inline src/function/scalar/string/prefix.cpp:92]: need comment

File: src/function/scalar/string/prefix.cpp [hunk 2]
```diff
@@ -59,13 +65,74 @@ struct PrefixOperator {
...
+	if (constant.IsNull()) {
+		return FilterPropagateResult::FILTER_ALWAYS_FALSE;
+	}
+
+	auto prefix = StringValue::Get(constant);
+	if (prefix.empty()) {
+		return column_stats->CanHaveNull() ? FilterPropagateResult::NO_PRUNING_POSSIBLE
+		                                   : FilterPropagateResult::FILTER_ALWAYS_TRUE;
+	}
+	if (StringStats::HasMaxStringLength(*column_stats) &&
```
Comment (dentiny) [inline src/function/scalar/string/prefix.cpp]: need comment: first check we could skip everything, next check whether we could leverage min/max

File: src/function/scalar/string/prefix.cpp [hunk 3]
```diff
@@ -59,13 +65,74 @@ struct PrefixOperator {
...
+	    StringStats::MaxStringLength(*column_stats) < prefix.size()) {
```
Comment (dentiny) [inline src/function/scalar/string/prefix.cpp]: No need to read -- cannot satisfy
